### PR TITLE
Added GetInterfaceType to NMState pkg

### PR DESCRIPTION
**Pull request by evgenLevin**

### CODE REVIEW

File: pkg/nmstate/shared.go [hunk 1]
```diff
@@ -36,3 +39,30 @@ type Vf struct {
 	Trust      bool   `yaml:"trust"`
 	VlanID     int    `yaml:"vlan-id"`
 }
+
+// Bridge provides struct for the NMState Interface Ethernet  Bridge state object
```
Comment (kononovn) [inline pkg/nmstate/shared.go:43]: Please remove double space

File: pkg/nmstate/shared.go [hunk 2]
```diff
@@ -36,3 +39,30 @@ type Vf struct {
...
+// containing interface Bridge information.
+type Bridge struct {
+	Port []map[string]string `yaml:"port,omitempty"`
+}
+
+// LinkAggregation provides struct for the NMState Interface Ethernet  LinkAggregation state object
```
Comment (kononovn) [inline pkg/nmstate/shared.go:49]: Please remove 2 spaces

File: pkg/nmstate/shared.go [hunk 3]
```diff
@@ -36,3 +39,30 @@ type Vf struct {
...
+// containing interface LinkAggregation information.
+type LinkAggregation struct {
+	Mode    string                 `yaml:"mode"`
+	Options OptionsLinkAggregation `yaml:"options,omitempty"`
+	Port    []string               `yaml:"port,omitempty"`
+}
+
+// OptionsLinkAggregation provides struct for the NMState Interface Ethernet  LinkAggregation Options state object
```
Comment (kononovn) [inline pkg/nmstate/shared.go:57]: Please remove 2 spaces

File: pkg/nmstate/shared.go [hunk 4]
```diff
@@ -36,3 +39,30 @@ type Vf struct {
...
+// containing interface LinkAggregation Options information.
+type OptionsLinkAggregation struct {
+	Primary string `yaml:"primary,omitempty"`
+}
+
+// Vlan provides struct for the NMState Interface Ethernet  Vlan Options state object
```
Comment (kononovn) [inline pkg/nmstate/shared.go:63]: Please remove 2 spaces

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix few comments

**Review by kononovn — APPROVED** [review]
lgtm

**Comment by evgenLevin** [comment]
@cdvultur Thanks!!!

**Pull request merged**